stdlib/ballerina-database/src/main/java/org/ballerinalang/database/sql/SQLDataIterator.java

```diff
@@ -255,14 +437,54 @@ public BStruct generateNext() {
                     }
                 }
             }
-        }  catch (Throwable e) {
+        } catch (Throwable e) {
             throw new BallerinaException(
                     "error in retrieving next value for column: " + columnName + ": at index:" + index + ":" + e
                             .getMessage());
         }
         return bStruct;
     }
 
+    private void handleUnionTypeStructField(BStruct bStruct, BStructType.StructField[] structFields, int index,
+            int refRegIndex, BRefType value, int refType) {
+        if (isNillableType(structFields, index - 1, refType)) {
+            bStruct.setRefField(refRegIndex, value);
+        } else {
+            throw new BallerinaException("Union type contains more than 2 members or the members do not result in "
+                    + "an assignable nillable type");
+        }
+    }
+
+    private void handleNilToNonNilableFieldAssignment() {
+        throw new BallerinaException("Trying to assign a Nil value to a non-nillable field");
+    }
+
+    private void handleMismatchingFieldAssignment() {
+        throw new BallerinaException("Trying to assign to a mismatching field");
+    }
+
+    private boolean isNillableType(BStructType.StructField[] structFields, int index, int typeTag) {
+        List<BType> memberTypes = ((BUnionType) structFields[index].getFieldType())
+                .getMemberTypes();
+        if (memberTypes.size() != 2) {
+            return false;
+        }
+        boolean containsType = false;
+        boolean containsNil = false;
+        for (BType memberType : memberTypes) {
+            if (memberType.getTag() == TypeTags.UNION_TAG) {
+                return false;
```

Comment:
Why do we need to check union types here?

Reply:
But union type never get passed into this as typeTag and even if the ballerina code has a union type it will get validated from the below if case. So we don't need this.